### PR TITLE
[WIP] handle non-adoc files

### PR DESCRIPTION
**Pull request by manaswinidas**

### CODE REVIEW

File: src/main/resources/templates/start.adoc [hunk 1]
```diff
@@ -1 +1,2 @@
 ifdef::context[:parent-context: {context}]
+:assets-dir:{../assets}
```
Comment (LightGuard) [inline src/main/resources/templates/start.adoc:2]: I don't think this part is necessary

File: src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java [hunk 2]
```diff
@@ -165,6 +167,29 @@ private void writeModules(Configuration config) {
         }
     }
 
+    private void moveNonadoc(Configuration config) {
+        try{
+        Path assetsDir = Files.createDirectories(config.getOutputDirectory().toPath().resolve("assets"));
+        final String extension = ".asciidoc";
+        final File sourceDir = config.getSourceDirectory();
+        final File destinationDir = assetsDir.toFile();
+        File[] files = sourceDir.listFiles((File pathname) -> !(pathname.getName().endsWith(extension)));
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java:176]: This won't work for the other asciidoc file extensions. 

.asciidoc, .adoc, and .ad are all valid extensions.

Take a look at https://github.com/asciidoctor/asciidoctorj/blob/master/asciidoctorj-core/src/main/java/org/asciidoctor/jruby/AsciiDocDirectoryWalker.java for the regex they're using to find asciidoc files.

File: src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java [hunk 3]
```diff
@@ -165,6 +167,29 @@ private void writeModules(Configuration config) {
...
+            try {
+                Files.copy(sourcePath, destinationPath,StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                //moving file failed.
+                throw new RuntimeException(e);
+            }
+        }
+    }  catch (IOException e) {
+        throw new RuntimeException(e);
+    }
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java:190]: I think this could be simplified with streams.

Comment (LightGuard) [reply src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java:190]: We should probably also use the `Issues` class that @maxandersen added.

File: src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java [hunk 4]
```diff
@@ -165,6 +167,29 @@ private void writeModules(Configuration config) {
...
+    private void moveNonadoc(Configuration config) {
+        try{
+        Path assetsDir = Files.createDirectories(config.getOutputDirectory().toPath().resolve("assets"));
+        final String extension = ".asciidoc";
+        final File sourceDir = config.getSourceDirectory();
+        final File destinationDir = assetsDir.toFile();
+        File[] files = sourceDir.listFiles((File pathname) -> !(pathname.getName().endsWith(extension)));
+        for(File f : files ){
+            Path sourcePath      = Paths.get(sourceDir.getAbsolutePath()+"/"+f.getName());
+            Path destinationPath = Paths.get(destinationDir.getAbsolutePath()+"/"+f.getName());
```
[the remark this answers is not in the document]
Comment (manaswinidas) [reply src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java:179]: no @LightGuard

**Comment by LightGuard** [comment]
I took some of what you did and modified it.

**Pull request closed without merging**